client/src/components/date/datepicker.js

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useEffect } from 'react';
+import DayPicker, { DateUtils } from 'react-day-picker';
+import 'react-day-picker/lib/style.css';
+import { toast } from 'react-toastify';
+
+const ReactDayPicker = ({ handleUpdate, updating, nextDueDate, frequency }) => {
+  const [days, setDate] = useState([]);
+  const [disabledDay, setDisabledDay] = useState('');
+
+  useEffect(() => {
+    if (updating) {     
+      const newArray = updating.map((day) => new Date(day));
+      setDate(newArray);
+
+    
+      if (frequency === 'yearly' || frequency === 'daily') {
+        setDisabledDay({ before: updating[0], after: updating[0] });
+      }
+    }
+  }, [updating, nextDueDate, frequency]);
+  
+  const handleDayClick = (day, { selected }) => {
+    if (frequency === 'weekly') {
+      return;
+    }
+
+    const selectedDays = days.concat();
+    if (selected) {
+      const selectedIndex = selectedDays.findIndex(selectedDay => DateUtils.isSameDay(selectedDay, day));
+      selectedDays.splice(selectedIndex, 1);
+      setDisabledDay(false);
+    } else  if ((frequency === 'yearly' || frequency === 'daily') && (days.length !== 0)) {
```

Comment:
last conditional doesn't need parens. AND I think you can just do `days.length` since 0 is falsey unless we ever get a negative number...?

```suggestion
    } else  if ((frequency === 'yearly' || frequency === 'daily') && days.length) {
```